tests/ssl-opt.sh

```diff
@@ -656,14 +656,32 @@ fi
 # used by watchdog
 MAIN_PID="$$"
 
-# be more patient with valgrind
+# We use somewhat arbitrary delays for tests:
+# - how long do we wait for the server to start (when lsof not available)?
+# - how long do we wait for the client to finish?
+#   (not to check performance, just to avoid waiting indefinitely)
+# - how long do we wait after client exit for completion of server tasks?
+#   (in particular, for it to finish writing & flushing its log file)
+# Things are slower with valgrind, so give extra time here.
+#
+# Note: there is a trade-off between the running time of this script and the
+# risk of spurious errors because we didn't wait long enough. In a better
+# world, we would be able to do without those delays (except the watchdog).
 if [ "$MEMCHECK" -gt 0 ]; then
-    START_DELAY=3
-    DOG_DELAY=30
+    START_DELAY=6
+    DOG_DELAY=60
+    SRV_DELAY_BASE=2
 else
-    START_DELAY=1
-    DOG_DELAY=10
+    START_DELAY=2
+    DOG_DELAY=15
```

Comment:
Why 15 and not 20?

Reply:
Because I was under the impression that the only spurious timeout we had so far where with valgrind, which would tend to indicate 10 was already enough, so I though +50% was enough here.

Now that you mention it, this has no impact on the normal running time of the script (by normal, I mean the client never hangs), so we could actually make the value much larger (like 1 min normal, 2 mins valgrind) just to be super-safe. (The only impact is that if the client is broken in such a way that it always hangs, then the running time of `ssl-opt.sh` will be around 300 * `$DOG_DELAY`, but this should probably be addressed at another level in the CI.)